assembly/lib.ts

```diff
@@ -4,33 +4,35 @@ import {
   ext_scratch_read,
   ext_scratch_size,
   ext_scratch_write,
-  ext_set_rent_allowance, 
+  ext_set_rent_allowance,
   ext_set_storage
 } from './env';
 
-enum Storage {
-  HAS_VALUE,
-  NO_VALUE
+export enum StorageResult {
+  Value,
+  None
 }
 
-// See https://stackoverflow.com/questions/15761790/convert-a-32bit-integer-into-4-bytes-of-data-in-javascript
-export function u32ToU8a (num: u32): Uint8Array {
-  const arr = new ArrayBuffer(4); // an u32 takes 4 bytes
-  const view = new DataView(arr);
-  view.setUint32(0, num, true); // byteOffset = 0; litteEndian = true
-  return Uint8Array.wrap(arr);
+export function toBytes<T>(num: T, le = true): Uint8Array {
+  // accept only integers and booleans
+  if (isInteger<T>()) {
+    const arr = new Uint8Array(sizeof<T>());
+    store<T>(arr.dataStart, le ? num : bswap(num));
+    return arr;
+  }
+  assert(false);
```

Comment:
The rewrite to `toBytes()` definitely makes it more generic and easier to re-use this function as a util for other contracts, thanks for that!